### ISSUE TITLE
Enable eslint rule "consistent-return"

### ISSUE THREAD

**Issue by samreid** [phetsims/chipper#1033]
From https://github.com/phetsims/chipper/issues/814, we would like to enable the rule "consistent-return".  Here is that discussion and consensus:

> MK: I could go either way on this.
> SR: I recommend enabling this rule. The existing occurrences look buggy.
> Dev team: Yes, assuming this doesn't affect cases where no return value was expected

There are 77 failures.  I'll work on it in this issue and assign other developers as necessary.

**Comment by samreid** [comment]
I enabled the rule.  There were 25+ cases where I think the responsible developer will need to take a look and address it, I've marked those cases with `// eslint-disable-line consistent-return`.  There are cases in these repos:
- [ ] perennial @mattpen @jonathanolson 
- [ ] phet-io-wrappers @zepumph 
- [ ] rosetta @jbphet 
- [ ] scenery @jonathanolson 
- [ ] studio @zepumph 
- [ ] weddell @jbphet 

There may be a few others, but let's assign to responsible devs for these repos and see how far we get.

**Comment by jbphet** [comment]
I've added @liam-mulhall and assigned him to the portions for rosetta.

@samreid - I'm not seeing any occurrences of `eslint-disable-line consistent-return` in the code for weddell.  There are plenty of other disabled lines, but I'm not seeing this particular one.  Am I missing something?

**Comment by samreid** [comment]
Please check https://github.com/phetsims/weddell/blob/b53dc1ad89025d45e4dfb6f3591f950e0e48a687/js/utils/notarize.js#L22-L27, is it possible weddell is excluded from your search?

**Comment by jonathanolson** [comment]
I've handled scenery. I'm not familiar with the perennial cases, I'll defer to @mattpen and @zepumph.

**Comment by zepumph** [comment]
I'll take a look and do all I can.

**Comment by zepumph** [comment]
It'd be good to have @mattpen double check my usages of `Promise.resolve()` in perennial.

**Comment by mattpen** [comment]
@zepumph - What you wrote should work, but I thought we should be using this pattern now:

```js
if ( errorCondition ) {
    throw someError;
}
else {
    return result;
}
// else block can be skipped if the function is "void"
```

This should be equivalent to the following and it should pass the lint rule.

```js
if ( errorCondition ) {
    return Promise.reject( someError );
}
else {
    return Promise.resolve( someError );
}
// else block can be skipped if the function is "void"
```

**Comment by jbphet** [comment]
I fixed the issue in weddell by explicitly returning `null` instead of just using an empty `return` statement.  This resolves the eslint issue and seems unlikely to cause any problems with notarizing, since it can't be done on non-Mac platforms anyway.

**Comment by zepumph** [comment]
@mattpen, https://github.com/phetsims/chipper/issues/1033#issuecomment-871653310 is a bit confusing to me, do you recommend changes from what I did in https://github.com/phetsims/perennial/commit/80a0610c06eca02c8d5cab664d4eb590bab2a5fe? I couldn't find any changes that went against your thoughts.

**Comment by mattpen** [comment]
@zepumph - sorry, reading it again it is not very clear.  Let me try again.  An async function is syntactic sugar for a Promise.  If you are in an async function and you return `Promise.resolve` or `Promise.reject`, you are essentially duplicating that call. Instead of `return Promise.reject(err)` you should instead `throw err` or `throw new Error( err )`, and instead of `return Promise.resolve(someValue)` you should `return someValue`.

Promise.resolve and Promise.reject are shorthand for returning promises in a syncronous function. While they might work in an async/await context, they complicate otherwise simple code.  These files illustrate some of the pains I went through when trying to learn this during chipper 2.0 and unfortunately I never cleaned them up.

Ping me if you have questions.

Maybe this comment isn't appropriate in this issue, since it isn't related to the consistent-return rule specifically.  Perhaps there should be a lint rule to the effect of "don't return a Promise in an async function".

To illustrate, here's a patch for your commit:

<details>

```js
Index: js/build-server/taskWorker.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/build-server/taskWorker.js b/js/build-server/taskWorker.js
--- a/js/build-server/taskWorker.js	(revision 80a0610c06eca02c8d5cab664d4eb590bab2a5fe)
+++ b/js/build-server/taskWorker.js	(date 1632263879414)
@@ -31,7 +31,7 @@
   winston.log( 'error', `BUILD ABORTED! ${err}` );
   winston.log( 'info', 'build aborted: checking out master for every repo in case build shas are still checked out' );
   await execute( 'grunt', [ 'checkout-master-all' ], constants.PERENNIAL );
-  return Promise.reject( new Error( `Build aborted, ${err}` ) );
+  throw new Error( `Build aborted, ${err}` );
 };
 
 /**
@@ -43,9 +43,8 @@
     await execute( 'rm', [ '-rf', buildDir ], '.' );
   }
   catch( err ) {
-    return abortBuild( err );
+    await abortBuild( err );
   }
-  return Promise.resolve();
 };
 
 /**
@@ -71,12 +70,12 @@
       await gitCheckout( 'chipper', 'master' );
       await gitPull( 'chipper' );
       winston.info( 'Deploy images completed successfully.' );
-      return Promise.resolve();
+      return;
     }
     catch( e ) {
       winston.error( e );
       winston.error( 'Deploy images failed. See previous logs for details.' );
-      return Promise.reject( e );
+      throw e;
     }
 
   }
@@ -114,29 +113,29 @@
 
         // make sure all keys in repos object are valid sim names
         if ( !simNameRegex.test( key ) ) {
-          return abortBuild( `invalid simName in repos: ${simName}` );
+          await abortBuild( `invalid simName in repos: ${simName}` );
         }
 
         const value = repos[ key ];
         if ( key === 'comment' ) {
           if ( typeof value !== 'string' ) {
-            return abortBuild( 'invalid comment in repos: should be a string' );
+            await abortBuild( 'invalid comment in repos: should be a string' );
           }
         }
         else if ( value instanceof Object && value.hasOwnProperty( 'sha' ) ) {
           if ( !/^[a-f0-9]{40}$/.test( value.sha ) ) {
-            return abortBuild( `invalid sha in repos. key: ${key} value: ${value} sha: ${value.sha}` );
+            await abortBuild( `invalid sha in repos. key: ${key} value: ${value} sha: ${value.sha}` );
           }
         }
         else {
-          return abortBuild( `invalid item in repos. key: ${key} value: ${value}` );
+          await abortBuild( `invalid item in repos. key: ${key} value: ${value}` );
         }
       }
     }
 
     // validate simName
     if ( !simNameRegex.test( simName ) ) {
-      return abortBuild( `invalid simName ${simName}` );
+      await abortBuild( `invalid simName ${simName}` );
     }
 
     // Infer brand from version string and keep unstripped version for phet-io
@@ -157,7 +156,7 @@
         winston.log( 'info', `detecting version number: ${version}` );
       }
       else {
-        return abortBuild( `invalid version number: ${version}` );
+        await abortBuild( `invalid version number: ${version}` );
       }
     }
 
@@ -165,7 +164,7 @@
     winston.log( 'info', `building sim ${simName}` );
 
     // Create the temporary build dir, removing the existing dir if it exists.
-    if ( await fs.existsSync( buildDir ) ) {
+    if ( fs.existsSync( buildDir ) ) {
       await execute( 'rm', [ '-rf', buildDir ], '.' );
     }
     await fs.promises.mkdir( buildDir, { recursive: true } );
@@ -207,7 +206,7 @@
       await execute( 'grunt', args, simDir );
     }
     else {
-      return abortBuild( 'Unsupported chipper version' );
+      await abortBuild( 'Unsupported chipper version' );
     }
 
     winston.debug( `deploying to servers: ${JSON.stringify( servers )}` );
@@ -357,10 +356,10 @@
     }
   }
   catch( err ) {
-    return abortBuild( err );
+    await abortBuild( err );
   }
 
-  return afterDeploy();
+  await afterDeploy();
 }
 
 module.exports = ( task, taskCallback ) => {
Index: js/build-server/pullMaster.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/build-server/pullMaster.js b/js/build-server/pullMaster.js
--- a/js/build-server/pullMaster.js	(revision 80a0610c06eca02c8d5cab664d4eb590bab2a5fe)
+++ b/js/build-server/pullMaster.js	(date 1632264011817)
@@ -27,7 +27,8 @@
     await gitPull( 'babel' );
   }
   catch( error ) {
-    return Promise.reject( new Error( 'git checkout/pull failed in babel' ) );
+    console.error( error );
+    throw new Error( 'git checkout/pull failed in babel' );
   }
 
   for ( const repoName in reposCopy ) {
@@ -38,7 +39,8 @@
       await execute( 'git', [ 'checkout', 'master' ], repoDir );
     }
     catch( error ) {
-      return Promise.reject( new Error( `git checkout master failed in ${repoName}` ) );
+      console.error( error );
+      throw new Error( `git checkout master failed in ${repoName}` );
     }
 
     try {
@@ -50,9 +52,6 @@
   }
 
   if ( errors.length > 0 ) {
-    return Promise.reject( new Error( 'at least one repository failed to pull master' ) );
-  }
-  else {
-    return Promise.resolve();
+    throw new Error( 'at least one repository failed to pull master' );
   }
 };
\ No newline at end of file
Index: js/common/writePhetioHtaccess.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/common/writePhetioHtaccess.js b/js/common/writePhetioHtaccess.js
--- a/js/common/writePhetioHtaccess.js	(revision 80a0610c06eca02c8d5cab664d4eb590bab2a5fe)
+++ b/js/common/writePhetioHtaccess.js	(date 1632264011809)
@@ -45,18 +45,13 @@
       latestRedirectContents += 'RewriteCond %{QUERY_STRING} =download\n' +
                                 'RewriteRule ([^/]*)$ - [L,E=download:$1]\n' +
                                 'Header onsuccess set Content-disposition "attachment; filename=%{download}e" env=download\n';
-      try {
-        await writeFile( redirectFilepath, latestRedirectContents );
-      }
-      catch( err ) {
-        return Promise.reject( err );
-      }
+      await writeFile( redirectFilepath, latestRedirectContents );
     }
     else {
       winston.error( `simName: ${latestOption.simName}` );
       winston.error( `version: ${latestOption.version}` );
       winston.error( `directory: ${latestOption.directory}` );
-      return Promise.reject( new Error( 'latestOption is missing one of the required parameters (simName, version, or directory)' ) );
+      throw new Error( 'latestOption is missing one of the required parameters (simName, version, or directory)' );
     }
   }
 
@@ -130,9 +125,7 @@
     }
     catch( err ) {
       winston.debug( 'phetio authentication htaccess not written' );
-      return Promise.reject( err );
+      throw err;
     }
   }
-
-  return Promise.resolve();
 };
```

</details>

**Comment by zepumph** [comment]
That makes a bunch of sense. Thanks for the patch. Applied, rule has been on in master, lint passing, and closing.
